omnilib/prototypes/recipe-generation.lua

```diff
@@ -1193,7 +1184,7 @@ function RecGen:import(recipe)
 			setPlace(proto.place_result):
 			setSubgroup(proto.subgroup):
 			setFuelCategory(proto.fuel_category):
-			setIcons(proto.icons or proto.icon):
+			setIcons(find_result_icon(proto)):
```

Comment:
Breaks recipe icons completely, if a recipe has icons we want to use them and not the icons of a result, so this should be icons or icon OR function to only use result icons in the worst case